src/Dfc.CourseDirectory.WebV2/Features/Venues/AddVenue/AddVenueController.cs

```diff
@@ -0,0 +1,193 @@
+﻿using System.Linq;
+using System.Threading.Tasks;
+using Flurl;
+using FormFlow;
+using MediatR;
+using Microsoft.AspNetCore.Mvc;
+
+namespace Dfc.CourseDirectory.WebV2.Features.Venues.AddVenue
+{
+    [Route("venues/add")]
+    [JourneyMetadata(
+        journeyName: "AddVenue",
+        stateType: typeof(AddVenueJourneyModel),
+        appendUniqueKey: true,
+        requestDataKeys: "providerId?")]
+    [RequireProviderContext]
+    public class AddVenueController : Controller
+    {
+        private const string ReturnUrlJourneyInstancePropertyKey = "returnUrl";
+
+        private readonly IMediator _mediator;
+        private readonly JourneyInstanceProvider _journeyInstanceProvider;
+        private readonly IProviderContextProvider _providerContextProvider;
+        private JourneyInstance<AddVenueJourneyModel> _journeyInstance;
```

Comment:
I'm not a fan of declaring `JourneyInstance` at the class level. While it's used across multiple methods in the class, it's better suited to being scoped as a local variable as that's its true intent.

In other words, it doesn't **need** to be here, so it shouldn't be.